contrib/DC-GAN.ipynb

```diff
@@ -0,0 +1,562 @@
+    "        x = F.leaky_relu(F.max_pool2d(self.conv1(x), 2), 0.01)\n",
+    "        x = self.bn1(x)\n",
+    "        x = F.leaky_relu(F.max_pool2d(self.conv2(x), 2), 0.01)\n",
+    "        x = self.bn2(x)\n",
+    "        x = torch.flatten(x, 1)\n",
+    "        x = F.leaky_relu(self.fc1(x), 0.01)\n",
+    "        return torch.sigmoid(x)            \n",
+    "        "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "rATu6mC2Qhxa"
+   },
+   "outputs": [],
+   "source": [
+    "class GenerativeNet(torch.nn.Module):\n",
+    "    \n",
+    "    def __init__(self):\n",
+    "        super(GenerativeNet, self).__init__()\n",
+    "        self.linear1 = nn.Linear(100, 1024) # relue activation\n",
+    "        self.bn1 = nn.BatchNorm1d(1024)\n",
+    "        self.linear2 = nn.Linear(1024, 7*7*128) # relu activation\n",
+    "        self.bn2 = nn.BatchNorm1d(7*7*128) # reshape to (-1, 7, 7, 128)\n",
+    "        self.conv1 = nn.ConvTranspose2d(\n",
+    "            in_channels=128, \n",
+    "            out_channels=64, \n",
+    "            kernel_size=4,\n",
+    "            stride=2, \n",
+    "            padding=1, \n",
+    "            bias=False\n",
+    "        ) #relu activation\n",
+    "        self.bn3 = nn.BatchNorm2d(64)\n",
+    "        self.conv2 = nn.ConvTranspose2d(\n",
+    "            in_channels=64, \n",
+    "            out_channels=1, \n",
+    "            kernel_size=4,\n",
+    "            stride=2, \n",
+    "            padding=1, \n",
+    "            bias=False\n",
+    "        ) # tanh activation\n",
+    "              \n",
+    "    def forward(self, x):\n",
+    "        x = self.linear1(x)\n",
+    "        x = F.relu(x)\n",
+    "        x = self.bn1(x)\n",
+    "        x = self.linear2(x)\n",
+    "        x = F.relu(x)\n",
+    "        x = self.bn2(x)\n",
+    "        x = x.view(x.shape[0], 128, 7, 7)\n",
+    "        x = self.conv1(x)\n",
+    "        x = F.relu(x)\n",
+    "        x = self.bn3(x)\n",
+    "        x = self.conv2(x)\n",
+    "        x = torch.tanh(x)\n",
+    "        return x"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "3oBrhFUEQhxc"
+   },
+   "outputs": [],
+   "source": [
+    "def init_weights(m):\n",
+    "    classname = m.__class__.__name__\n",
+    "    if classname.find('Conv') != -1 or classname.find('BatchNorm') != -1:\n",
+    "        m.weight.data.normal_(0.00, 0.02)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "EZ45r5UvQhxe"
+   },
+   "outputs": [],
+   "source": [
+    "def real_data_target(size, device):\n",
+    "    '''\n",
+    "    Tensor containing ones, with shape = size\n",
+    "    '''\n",
+    "    data = Variable(torch.ones(size, 1))\n",
```

Comment:
The `Variable` syntax is deprecated:

https://pytorch.org/docs/stable/autograd.html#variable-deprecated

Please che other usage around this notebook.

Reply:
Fixed.